pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -114,9 +114,18 @@ func (c *controller) RegisterEventHandler(typ string, f func(model.Config, model
 
 		// TODO: This works well for Add and Delete events, but not so for Update:
 		// An updated ingress may also trigger an Add or Delete for one of its constituent sub-rules.
-		rules := convertIngress(*ingress, c.domainSuffix)
-		for _, rule := range rules {
-			f(rule, event)
+		switch typ {
```

Comment:
the logic for adding event handlers for virtualservice/gateway configs was missing.